sensing/src/test/java/com/google/android/sensing/impl/SensorManagerImplTest.kt

```diff
@@ -0,0 +1,227 @@
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.sensing.impl
+
+import android.content.Context
+import android.os.Build
+import androidx.lifecycle.testing.TestLifecycleOwner
+import androidx.test.core.app.ApplicationProvider
+import com.google.android.sensing.SensingEngine
+import com.google.android.sensing.SensingEngineConfiguration
+import com.google.android.sensing.SensorManager
+import com.google.android.sensing.model.CaptureInfo
+import com.google.android.sensing.testing.TEST_EXTERNAL_ID
+import com.google.android.sensing.testing.TEST_OUTPUT_FOLDER
+import com.google.android.sensing.testing.TEST_SENSOR_FACTORY
+import com.google.android.sensing.testing.TEST_SENSOR_TYPE
+import com.google.android.sensing.testing.TestApplication
+import com.google.android.sensing.testing.TestSensorCaptureRequest
+import com.google.android.sensing.testing.TestSensorInitConfig
+import kotlinx.coroutines.test.runTest
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+/** Unit tests for [SensorManagerImpl]. */
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [Build.VERSION_CODES.P], application = TestApplication::class)
+class SensorManagerImplTest {
+  private val sensorManager =
+    SensorManager.getInstance(ApplicationProvider.getApplicationContext<TestApplication>())
+
+  private val sensingEngine =
+    SensingEngine.getInstance(ApplicationProvider.getApplicationContext<TestApplication>())
+
+  @Before
+  fun beforeSetup() {
+    check(
+      ApplicationProvider.getApplicationContext<TestApplication>()
+        is SensingEngineConfiguration.Provider
+    ) {
+      "Few tests require a custom application class that implements SensingEngineConfiguration.Provider"
+    }
+  }
+
+  @After
+  fun afterSetup() {
+    sensorManager.reset(TEST_SENSOR_TYPE)
+    sensorManager.unregisterSensorFactory(TEST_SENSOR_TYPE)
+  }
+
+  @Test
+  fun registerSensorFactory_checkRegistration_shouldBeTrue() {
+    sensorManager.registerSensorFactory(TEST_SENSOR_TYPE, TEST_SENSOR_FACTORY)
+
+    val checkRegistration = sensorManager.checkRegistration(TEST_SENSOR_TYPE)
+
+    assert(checkRegistration)
+  }
+
+  @Test
+  fun registerSensorFactory_reRegister_shouldGiveIllegalStateException() {
+    sensorManager.registerSensorFactory(TEST_SENSOR_TYPE, TEST_SENSOR_FACTORY)
+
+    try {
+      sensorManager.registerSensorFactory(TEST_SENSOR_TYPE, TEST_SENSOR_FACTORY)
+    } catch (e: Exception) {
+      assert(e is IllegalStateException)
+    }
```

Comment:
```suggestion

assertFailsWith<IllegalStateException> {
sensorManager.registerSensorFactory(TEST_SENSOR_TYPE, TEST_SENSOR_FACTORY)
}
```
[assertFailsWith](https://kotlinlang.org/api/latest/kotlin.test/kotlin.test/assert-fails-with.html)

Reply:
done